hank-core/src/main/java/com/liveramp/hank/zookeeper/ZooKeeperPlus.java

```diff
@@ -375,12 +375,26 @@ public void ensureCreated(String path, byte[] value) throws InterruptedException
     }
 
     public void ensureCreated(String path, byte[] value, CreateMode createMode) throws InterruptedException, KeeperException {
-      if (exists(path, false) == null) {
+      if (!path.isEmpty() && exists(path, false) == null) {
+        ensureCreated(getParent(path), null, createMode);
         create(path, value, DEFAULT_ACL, createMode);
         NodeCreationBarrier.block(ZooKeeperPlus.this, path);
       }
     }
 
+    private boolean isRootOrEmpty(String path){
+      return path.isEmpty() || path.equals("/");
```

Comment:
I don't think it munges unless you do anything that explicitly asks it to. Could be wrong, but probably cleaner than doing the string parsing ourselves.

There's also `Path`. Think it has fewer munge-with-filesystem methods. Maybe it'd be more appropriate.